units/github_status_api.go

```diff
@@ -0,0 +1,316 @@
+package units
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model/admin"
+	"github.com/evergreen-ci/evergreen/model/build"
+	"github.com/evergreen-ci/evergreen/model/patch"
+	"github.com/google/go-github/github"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/mongodb/grip/sometimes"
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2"
+)
+
+const (
+	githubStatusUpdateJobName = "github-status-update"
+	githubStatusError         = "error"
+	githubStatusFailure       = "failure"
+	githubStatusPending       = "pending"
+	githubStatusSuccess       = "success"
+
+	githubUpdateTypeBuild = "build"
+	githubUpdateTypePatch = "patch"
+)
+
+func init() {
+	registry.AddJobType(githubStatusUpdateJobName, func() amboy.Job { return makeGithubStatusUpdateJob() })
+}
+
+type githubStatus struct {
+	owner    string
+	repo     string
+	prNumber int
+	ref      string
+
+	urlPath     string
+	description string
+	context     string
+	ghStatus    string
+}
+
+func (status *githubStatus) Valid() bool {
+	if status.owner == "" || status.repo == "" || status.prNumber == 0 ||
+		status.ref == "" || status.urlPath == "" || status.context == "" {
+		return false
+	}
+	if status.ghStatus != githubStatusError && status.ghStatus != githubStatusFailure &&
+		status.ghStatus != githubStatusPending && status.ghStatus != githubStatusSuccess {
+		return false
+	}
```

Comment:
nit: I'd make these case statements, which I think would be easier to read

Reply:
done